client/state/data-layer/wpcom/plans/index.js

```diff
@@ -23,15 +26,19 @@ import {
  * @returns {Object} original action
  */
 export const requestPlans = ( { dispatch }, action, next ) => {
-	dispatch( http( {
+	const httpAction = http( {
 		apiVersion: '1.4',
 		method: 'GET',
 		path: '/plans',
 		onSuccess: action,
 		onFailure: action,
-	} ) );
+	} );
+
+	dispatch( httpAction );
 
-	return next( action );
+	if ( httpAction.type !== WPCOM_HTTP_BAD_REQUEST ) {
+		next( action );
+	}
```

Comment:
this code here is one of the reasons I feel very strongly against creating a Redux action for bad requests. A request which is always invalid and we can tell at compile-time should never be checked for in other parts of the code otherwise we'll end up with hundreds of redundant checks that aren't needed and pollute other code. Let's kill the error at the source instead of propagating it around the app.